crates/red_knot_python_semantic/src/types.rs

```diff
@@ -658,3 +665,9 @@ pub struct BytesLiteralType<'db> {
     #[return_ref]
     value: Box<[u8]>,
 }
+
+#[salsa::interned]
+pub struct TupleLiteralType<'db> {
+    #[return_ref]
+    elements: Vec<Type<'db>>,
```

Comment:
Since this should be immutable once constructed, and we will have a lot of these and they'll be long-lived, do you think it makes sense to use a boxed slice instead?

Reply:
Yeah, make sense, thanks